sdk/typespec/typespec_client_core/src/base64.rs

```diff
@@ -51,3 +52,50 @@ where
 {
     Ok(URL_SAFE.decode(input)?)
 }
+
+pub fn deserialize<'de, D>(deserializer: D) -> Result<Option<Vec<u8>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let decoded = <Option<String>>::deserialize(deserializer)?;
+    match decoded {
+        Some(d) => {
+            let d = decode(d).map_err(serde::de::Error::custom)?;
+            Ok(Some(d))
+        }
+        None => Ok(None),
+    }
+}
+
+pub fn deserialize_url_safe<'de, D>(deserializer: D) -> Result<Option<Vec<u8>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let decoded = <Option<String>>::deserialize(deserializer)?;
+    match decoded {
+        Some(d) => {
+            let d = decode_url_safe(d).map_err(serde::de::Error::custom)?;
+            Ok(Some(d))
+        }
+        None => Ok(None),
+    }
+}
+
+pub fn serialize<S>(to_serialize: &Option<Vec<u8>>, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let encoded = to_serialize.as_ref().map(encode);
+    <Option<String>>::serialize(&encoded, serializer)
+}
+
+pub fn serialize_url_safe<S>(
```

Comment:
Does declaring a `T` as `AsRef<[u8]>` not work here like the `serialize` functions above? More flexible.

Reply:
Would the param be an `Option<T>` or just a `T`?

Reply:
Ideally, we'd have one with and without `Option<T>`, but we don't really need one sans `Option` so I wouldn't worry about that. All our model fields are optional and we can always add non-optional support later.

Reply:
Fixed.